Some code/tests are racy

#### What versions are you running?

<pre>
$ git log -1
commit 4fa78c6cac9c1628fb3422758ee1df2bd1ab0c94 (HEAD -> master, tag: v0.6.1, origin/master, origin/HEAD)
Author: Kenneth Shaw <kenshaw@gmail.com>
Date:   Wed Jan 13 11:34:08 2021 +0700

    Updating cdproto dependency

$ chromium-browser --version
Chromium 87.0.4280.88 Fedora Project

$ go version
go version go1.15.6 linux/amd64
</pre>


#### What did you do? Include clear steps.
$ git clone https://github.com/chromedp/chromedp
$ cd chromedp
$ go test -race


#### What did you expect to see?
Tests pass without races.


#### What did you see instead?
```
==================
WARNING: DATA RACE
Read at 0x00c000df8540 by goroutine 56:
  runtime.mapaccess1_fast64()
      /usr/lib/golang/src/runtime/map_fast64.go:12 +0x0
  github.com/chromedp/chromedp.(*Target).enclosingFrame()
      /home/elliott/code/chromedp/target.go:59 +0x15e
  github.com/chromedp/chromedp.(*Selector).Do()
      /home/elliott/code/chromedp/query.go:191 +0x3cc
  github.com/chromedp/chromedp.Tasks.Do()
      /home/elliott/code/chromedp/chromedp.go:614 +0x8e
  github.com/chromedp/chromedp.Run()
      /home/elliott/code/chromedp/chromedp.go:272 +0x164
  github.com/chromedp/chromedp.TestQueryIframe()
      /home/elliott/code/chromedp/nav_test.go:290 +0x9a8
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1123 +0x202

Previous write at 0x00c000df8540 by goroutine 181:
  runtime.mapassign_fast64()
      /usr/lib/golang/src/runtime/map_fast64.go:92 +0x0
  github.com/chromedp/chromedp.walk()
      /home/elliott/code/chromedp/util.go:94 +0xdb
  github.com/chromedp/chromedp.walk()
      /home/elliott/code/chromedp/util.go:102 +0x175
  github.com/chromedp/chromedp.walk()
      /home/elliott/code/chromedp/util.go:102 +0x175
  github.com/chromedp/chromedp.setChildNodes.func1()
      /home/elliott/code/chromedp/util.go:143 +0xe4
  github.com/chromedp/chromedp.(*Target).domEvent()
      /home/elliott/code/chromedp/target.go:404 +0x269
  github.com/chromedp/chromedp.(*Target).run()
      /home/elliott/code/chromedp/target.go:129 +0x267

Goroutine 56 (running) created at:
  testing.(*T).Run()
      /usr/lib/golang/src/testing/testing.go:1168 +0x5bb
  testing.runTests.func1()
      /usr/lib/golang/src/testing/testing.go:1439 +0xa6
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1123 +0x202
  testing.runTests()
      /usr/lib/golang/src/testing/testing.go:1437 +0x612
  testing.(*M).Run()
      /usr/lib/golang/src/testing/testing.go:1345 +0x3b3
  github.com/chromedp/chromedp.TestMain()
      /home/elliott/code/chromedp/chromedp_test.go:88 +0x234
  main.main()
      _testmain.go:261 +0x271

Goroutine 181 (running) created at:
  github.com/chromedp/chromedp.(*Context).attachTarget()
      /home/elliott/code/chromedp/chromedp.go:353 +0x4f1
  github.com/chromedp/chromedp.(*Context).newTarget()
      /home/elliott/code/chromedp/chromedp.go:303 +0xcd4
  github.com/chromedp/chromedp.Run()
      /home/elliott/code/chromedp/chromedp.go:268 +0x1ca
  github.com/chromedp/chromedp.testAllocate()
      /home/elliott/code/chromedp/chromedp_test.go:118 +0x324
  github.com/chromedp/chromedp.TestQueryIframe()
      /home/elliott/code/chromedp/nav_test.go:276 +0x87
  testing.tRunner()
      /usr/lib/golang/src/testing/testing.go:1123 +0x202
==================
--- FAIL: TestQueryIframe (0.15s)
    testing.go:1038: race detected during execution of test
--- FAIL: TestWithBrowserOptionAlreadyAllocated (0.19s)
    testing.go:1038: race detected during execution of test
--- FAIL: TestRemoteAllocator (0.37s)
    testing.go:1038: race detected during execution of test
--- FAIL: TestMatchedStyle (0.00s)
--- FAIL: TestComputedStyle (0.00s)
--- FAIL: TestKeyEvent (0.00s)
--- FAIL: TestSubmit (0.00s)
--- FAIL: TestMouseClickOffscreenNode (0.00s)
--- FAIL: TestKeyEventNode (0.00s)
--- FAIL: TestSendKeys (0.00s)
--- FAIL: TestMouseClickNode (0.00s)
--- FAIL: TestCloseDialog (0.00s)
--- FAIL: TestClick (0.00s)
--- FAIL: TestDoubleClick (0.00s)
--- FAIL: TestRunResponse (0.01s)
--- FAIL: TestRemoveAttribute (0.00s)
--- FAIL: TestSetAttributeValue (0.00s)
--- FAIL: TestSetValue (0.00s)
--- FAIL: TestSetAttributes (0.00s)
--- FAIL: TestSVGFullXPath (0.00s)
--- FAIL: TestReset (0.00s)
--- FAIL: TestDialTimeout (0.00s)
--- FAIL: TestClear (0.00s)
```

At this point, it hangs, possibly due to the data race, possibly due to #675.

--- a/target.go
+++ b/target.go
@@ -47,6 +47,8 @@ func (t *Target) enclosingFrame(node *cdp.Node) cdp.FrameID {
 	t.frameMu.RLock()
 	top := t.frames[t.cur]
 	t.frameMu.RUnlock()
+	top.RLock()
+	defer top.RUnlock()
 	for {
 		if node == nil {
 			// Avoid crashing. This can happen if we're using an old

